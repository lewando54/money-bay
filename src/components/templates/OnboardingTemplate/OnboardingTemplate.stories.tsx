/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { Meta,StoryObj } from '@storybook/react-native'

import Images from '@assets/images'

import OnboardingTemplate from './OnboardingTemplate'

const meta = {
    component: OnboardingTemplate,
    argTypes: {
        data: { control: 'object' },
        onButtonPress: { action: 'buttonPressed' }
    }
} satisfies Meta<typeof OnboardingTemplate>

export default meta
type Story = StoryObj<typeof OnboardingTemplate>

export const Default: Story = {
    args: {
        data: [
            {
                id: 0,
                imageSource: Images.onboarding2,
                heading: 'Welcome to MoneyBay app!',
                body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
                testID: 'slide-0'
            },
            {
                id: 1,
                imageSource: Images.onboarding3,
                heading: 'Get a new card in a few clicks!',
                body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
                testID: 'slide-1'
            },
            {
                id: 2,
                imageSource: Images.onboarding4,
                heading: 'Easy payments all over the world!',
                body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
                testID: 'slide-2'
            }
        ],
    }
}
