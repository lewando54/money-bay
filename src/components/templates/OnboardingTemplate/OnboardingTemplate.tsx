import React from 'react'
import onboardingTemplateStyle from './OnboardingTemplate.style'
import OnboardingSlides, { TOnboardingSlides } from '../../organisms/OnboardingSlides/OnboardingSlides'
import { ImageBackground, View } from 'react-native'
import Images from 'assets/images'

interface IOnboardingTemplateProps {
    data: TOnboardingSlides
    onButtonPress?: () => void
    testID?: string
}

export default function OnboardingTemplate({
    data=[
        {
            id: 0,
            imageSource: Images.onboarding2,
            heading: 'Welcome to MoneyBay app!',
            body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
            testID: 'slide-0'
        },
        {
            id: 1,
            imageSource: Images.onboarding3,
            heading: 'Get a new card in a few clicks!',
            body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
            testID: 'slide-0'
        },
        {
            id: 2,
            imageSource: Images.onboarding4,
            heading: 'Easy payments all over the world!',
            body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
            testID: 'slide-0'
        }
    ],
    onButtonPress,
    testID
}: IOnboardingTemplateProps){
    return (
        <View testID={testID} style={{flex: 1}}>
            <ImageBackground style={{flex: 1}} resizeMode="cover" source={require('../../../../assets/img/bg-story.png')}>
                <OnboardingSlides data={data} testID={testID} onPress={onButtonPress}/>
            </ImageBackground>
        </View>
    )
}
