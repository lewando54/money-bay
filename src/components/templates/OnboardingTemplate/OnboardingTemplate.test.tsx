import React from 'react'
import renderer from 'react-test-renderer'
import Images from '@assets/images'
import OnboardingTemplate from './OnboardingTemplate'
describe('OnboardingTemplate', () => {
    const data=[
        {
            id: 0,
            imageSource: Images.onboarding2,
            heading: 'Welcome to MoneyBay app!',
            body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
            testID: 'slide-0'
        },
        {
            id: 1,
            imageSource: Images.onboarding3,
            heading: 'Get a new card in a few clicks!',
            body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
            testID: 'slide-0'
        },
        {
            id: 2,
            imageSource: Images.onboarding4,
            heading: 'Easy payments all over the world!',
            body: 'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
            testID: 'slide-0'
        }
    ]

    it('should render correctly', () => {
        const tree = renderer.create(<OnboardingTemplate data={data}/>)
        expect(tree).toMatchSnapshot()
    })
})
