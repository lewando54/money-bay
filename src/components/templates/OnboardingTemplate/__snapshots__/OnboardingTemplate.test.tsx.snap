// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingTemplate should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "backgroundColor": "white",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "bottom": 0,
          "gap": 45,
          "height": 667,
          "justifyContent": "flex-end",
          "left": 0,
          "marginHorizontal": 20,
          "padding": 20,
          "paddingBottom": 35.354281639999996,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": "auto",
            "marginRight": "auto",
            "width": 96,
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "opacity": 1,
              },
            }
          }
          collapsable={false}
          style={
            {
              "borderColor": "#1B1D4D",
              "borderRadius": 100,
              "borderWidth": 3,
              "height": 8,
              "opacity": 1,
              "width": 8,
            }
          }
        />
        <View
          animatedStyle={
            {
              "value": {
                "opacity": 0.2,
              },
            }
          }
          collapsable={false}
          style={
            {
              "borderColor": "#1B1D4D",
              "borderRadius": 100,
              "borderWidth": 3,
              "height": 8,
              "opacity": 0.2,
              "width": 8,
            }
          }
        />
        <View
          animatedStyle={
            {
              "value": {
                "opacity": 0.2,
              },
            }
          }
          collapsable={false}
          style={
            {
              "borderColor": "#1B1D4D",
              "borderRadius": 100,
              "borderWidth": 3,
              "height": 8,
              "opacity": 0.2,
              "width": 8,
            }
          }
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            {
              "colors": [
                4288142079,
                4279654301,
              ],
              "endPoint": [
                1,
                0.5,
              ],
              "locations": undefined,
              "startPoint": [
                0,
                0.5,
              ],
            }
          }
          style={
            [
              {
                "borderRadius": 10,
                "padding": 15,
              },
              false,
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                [
                  {
                    "fontFamily": "MulishSemiBold",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                  {
                    "color": "white",
                  },
                  false,
                  false,
                ],
              ]
            }
          >
            Get started
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
    <View>
      <RCTScrollView
        CellRendererComponent={[Function]}
        animatedStyle={
          {
            "value": {},
          }
        }
        collapsable={false}
        data={
          [
            {
              "body": "Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.",
              "heading": "Welcome to MoneyBay app!",
              "id": 0,
              "imageSource": 1,
              "testID": "slide-0",
            },
            {
              "body": "Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.",
              "heading": "Get a new card in a few clicks!",
              "id": 1,
              "imageSource": 1,
              "testID": "slide-0",
            },
            {
              "body": "Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.",
              "heading": "Easy payments all over the world!",
              "id": 2,
              "imageSource": 1,
              "testID": "slide-0",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        overScrollMode="auto"
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        snapToAlignment="center"
        stickyHeaderIndices={[]}
        viewabilityConfig={
          {
            "viewAreaCoveragePercentThreshold": 10,
          }
        }
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            animatedStyle={
              {
                "value": {},
              }
            }
            collapsable={false}
            onLayout={[Function]}
          >
            <View
              animatedStyle={
                {
                  "value": {
                    "opacity": 1,
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "height": 1027.18,
                  "justifyContent": "flex-start",
                  "marginTop": 106.72,
                  "opacity": 1,
                  "width": 750,
                }
              }
              testID="slide-0"
            >
              <Image
                source={1}
                style={
                  {
                    "height": 520.26,
                    "resizeMode": "contain",
                    "width": 675,
                  }
                }
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "gap": 20,
                    "overflow": "hidden",
                    "paddingHorizontal": 20,
                    "paddingTop": 80.03999999999999,
                    "width": 669.975,
                  }
                }
              >
                <SvgMock />
                <Text
                  style={
                    [
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                      },
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishBold",
                        "fontSize": 28,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Welcome to MoneyBay app!
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                      },
                      {
                        "color": "#4C4C60",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                        "paddingHorizontal": 6,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
                </Text>
              </View>
            </View>
          </View>
          <View
            animatedStyle={
              {
                "value": {},
              }
            }
            collapsable={false}
            onLayout={[Function]}
          >
            <View
              animatedStyle={
                {
                  "value": {
                    "opacity": 0,
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "height": 1027.18,
                  "justifyContent": "flex-start",
                  "marginTop": 106.72,
                  "opacity": 0,
                  "width": 750,
                }
              }
              testID="slide-0"
            >
              <Image
                source={1}
                style={
                  {
                    "height": 520.26,
                    "resizeMode": "contain",
                    "width": 675,
                  }
                }
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "gap": 20,
                    "overflow": "hidden",
                    "paddingHorizontal": 20,
                    "paddingTop": 80.03999999999999,
                    "width": 669.975,
                  }
                }
              >
                <SvgMock />
                <Text
                  style={
                    [
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                      },
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishBold",
                        "fontSize": 28,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Get a new card in a few clicks!
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                      },
                      {
                        "color": "#4C4C60",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                        "paddingHorizontal": 6,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
                </Text>
              </View>
            </View>
          </View>
          <View
            animatedStyle={
              {
                "value": {},
              }
            }
            collapsable={false}
            onLayout={[Function]}
          >
            <View
              animatedStyle={
                {
                  "value": {
                    "opacity": 0,
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "height": 1027.18,
                  "justifyContent": "flex-start",
                  "marginTop": 106.72,
                  "opacity": 0,
                  "width": 750,
                }
              }
              testID="slide-0"
            >
              <Image
                source={1}
                style={
                  {
                    "height": 520.26,
                    "resizeMode": "contain",
                    "width": 675,
                  }
                }
              />
              <View
                style={
                  {
                    "alignItems": "center",
                    "gap": 20,
                    "overflow": "hidden",
                    "paddingHorizontal": 20,
                    "paddingTop": 80.03999999999999,
                    "width": 669.975,
                  }
                }
              >
                <SvgMock />
                <Text
                  style={
                    [
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                      },
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishBold",
                        "fontSize": 28,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Easy payments all over the world!
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#1B1D4D",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                      },
                      {
                        "color": "#4C4C60",
                        "fontFamily": "MulishRegular",
                        "fontSize": 16,
                        "paddingHorizontal": 6,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
